pkg/datarepair/checker/checker.go

```diff
@@ -127,8 +141,19 @@ func (checker *Checker) IdentifyInjuredSegments(ctx context.Context) (err error)
 
 					// reset durability stats for next iteration
 					checker.monStats = durabilityStats{}
+
+					if checker.pieceTracker != nil {
+						err := checker.gcService.Send(ctx, checker.pieceTracker, func() {
+							checker.gcWaitGroup.Done()
```

Comment:
This control should be somewhere else and not through callbacks, these are hard to verify for correctness.